### ISSUE TITLE
date and date indications

### ISSUE THREAD

**Issue by eroux** [buda-base/xmltoldmigration#101]
Following [RFC017](https://docs.google.com/document/d/1MhEXs8hxBjYp6qXrmNIKGFyhMbKa53YOjhDJCeUdxTI/edit#) and [RFC018](https://docs.google.com/document/d/1dUO5Yb3Fs9yp_6Y1HGJ-Vw7nFghFuRwhYlowSln6QJo/edit), and https://github.com/buda-base/owl-schema/issues/149 ; I think we should:

- [ ] replace `xsd:integers` by `xsd:gYear` for gregorian years (note that this requires padding `0`s)
- [ ] replace `bdr:foo bdo:onYear "1543"^^bdo:beDate` with:

```ttl
bdr:foo bdo:dateIndication [
      :yearInEra 1543 ;
      :era bdr:EraBE 
   ] ;
   :notBefore "1000"^^xsd:gYear ;
   :notAfter "1001"^^xsd:gYear .
```

(replacing the blank node with a named URI)

**Comment by eroux** [comment]
or: 

```ttl
bdr:foo bdo:yearIndication "1543"^^bdo:beYearIndication .
```

**Comment by xristy** [comment]
or:

    bdr:foo bdo:beYear 1543 ;
               :notBefore "1000"^^xsd:gYear ;
               :notAfter "1001"^^xsd:gYear .

**Comment by xristy** [comment]
bdr:foo bdo:beDate "1543-10-23" .

**Comment by xristy** [comment]
bdr:foo bdo:beYear 1543 , bdo:ceYear 1000 ;

**Comment by eroux** [comment]
bdr:foo bdo:beYear 1543 ;
           :onYear "1000"^^xsd:gYear .

**Comment by xristy** [comment]
bdo:Date a union( bdo:Year , bdo:YearInterval , xsd:date , bdo:DateInterval ) .

    bdo:date : owl:Thing  ==>  bdo:Date .

    bdo:YearInterval has notBefore and notAfter

**Comment by xristy** [comment]
To recap with some examples. First, a common case from _legacy_ metadata taken from P314. Current:
```ttl
bdr:EV42E0D51B7002  a   bdo:PersonBirth ;
    bdo:eventWhere  bdr:G665 ;
    bdo:notAfter    1730 ;
    bdo:notBefore   1729 ;
    .
bdr:EV6186A8935F68  a  bdo:PersonDeath ;
    bdo:onYear  1798 ;
    .
```

New:
```ttl
bdr:EV42E0D51B7002  a   bdo:PersonBirth ;
    bdo:eventWhere  bdr:G665 ;
    bdo:notAfter    "1730"^^gYear ;
    bdo:notBefore   "1729"^^gYear ;
    .
bdr:EV6186A8935F68  a  bdo:PersonDeath ;
    bdo:onYear  "1798"^^gYear ;
    .
```

From W1FEMC010026, Current:
```ttl
bdr:EVA8FDDA4BA9FC  a  bdo:CompletedEvent ;
    bdo:onYear  "2500"^^bdo:beDate ;
    .
```

New:
```ttl
bdr:EVA8FDDA4BA9FC  a  bdo:CompletedEvent ;
    bdo:dateIndication  bdr:DI9A88CF70B ;
    bdo:notBefore "1957"^^xsd:gYear ;
    bdo:notAfter "1958"^^xsd:gYear ;
    .
bdr:DI9A88CF70B a bdo:DateIndication ;
    bdo:yearInEra 2500 ;
    bdo:era BuddhistEra ;
    .
bdo:BuddhistEra a owl:Class ;
    adm:userTooltip "Buddhist Era Year dating from 543 B.C.E."@en ;
    bdo:yearOffset -543 ;
    rdfs:label "B.E." ;
    rdfs:subClassOf bdo:Era ;
.
```
See the updated date_indications.ttl.

Agreed?

**Comment by eroux** [comment]
For P314 yes. Note that gYear must always have 4 digits (so it must be panned with 0s if necessary).

For W1FEMC010026 generally speaking yes. I've reorganized the eras a little bit so that it's more coherent with the Chinese eras, so it would be something like:

```ttl
bdr:EVA8FDDA4BA9FC  a  bdo:CompletedEvent ;
    bdo:dateIndication  bdr:DI9A88CF70B ;
    bdo:notBefore "1957"^^xsd:gYear ;
    bdo:notAfter "1958"^^xsd:gYear ;
    .
bdr:DI9A88CF70B a bdo:DateIndication ;
    bdo:yearInEra 2500 ;
    bdo:era bdr:EraBE ; # (that's the only change)
    .
```

**Comment by xristy** [comment]
fixed via [c6966ea](https://github.com/buda-base/xmltoldmigration/commit/c6966ea16343fb2cb334e45c3d1c7208283a6ccd), [b5e182d](https://github.com/buda-base/bdrc-libraries/commit/b5e182d63de097b35bda99c11b98ef31c0539849), and [db4c5ed](https://github.com/buda-base/xmltoldmigration/commit/db4c5ed38dfda785a14cd1a93bd0ebf5fc4646d4).

**Comment by eroux** [comment]
thanks!
